File: simple-stack-example-fragments/src/main/java/com/zhuinden/simplestackdemoexamplefragments/application/injection/NavigationModule.java
package com.zhuinden.simplestackdemoexamplefragments.application.injection;

import com.zhuinden.simplestack.Backstack;
import com.zhuinden.simplestackdemoexamplefragments.util.BackstackHolder;

import dagger.Module;
import dagger.Provides;

/**
 * Created by Owner on 2017. 01. 27..
 */

@Module
public class NavigationModule {
    @Provides
    public Backstack backstack(BackstackHolder backstackHolder) {
        return backstackHolder.getBackstack();
    }
}

File: simple-stack-example-mvp-fragments/src/main/java/com/zhuinden/simplestackdemoexamplefragments/application/injection/NavigationModule.kt
package com.zhuinden.simplestackdemoexamplefragments.application.injection

import com.zhuinden.simplestack.Backstack
import com.zhuinden.simplestackdemoexamplefragments.util.BackstackHolder

import dagger.Module
import dagger.Provides

/**
 * Created by Zhuinden on 2018. 08. 20.
 */

@Module
object NavigationModule {
    @Provides
    @JvmStatic
    fun backstack(backstackHolder: BackstackHolder): Backstack = backstackHolder.backstack
}